style-guide: add keypress documentation instructions. (#15798)
tldr-pages/tldr: contributing-guides/style-guide.md

- Proper names should be capitalized in the description whenever applicable (e.g. use `A tool for interacting with a Git repository.` instead of ``A tool for interacting with a `git` repository.``).
- Acronym expansions (i.e. protocols, tools, etc) must not be translated unless there is a recognized native equivalent for them.
- When documenting keycaps or a keyboard shortcut for a utility, make it stand out in the description:

1. If it is not translatable, enclose it with backticks (i.e. ``Print the last lines of a given file and keep reading it until `Ctrl + C`:``)

> 2. If it is translatable, enclose it with double angled brackets inside a placeholder (i.e. ``:wq{{<<Enter>>}}``).

Author: This should be reformatted.

Author: I tested with `grep -r "<<Enter>>". This pattern is used nowhere in the whole project.

- Proper names should be capitalized in the description whenever applicable (e.g. use `A tool for interacting with a Git repository.` instead of ``A tool for interacting with a `git` repository.``).
- Acronym expansions (i.e. protocols, tools, etc) must not be translated unless there is a recognized native equivalent for them.

> - When documenting keycaps or a keyboard shortcut for a utility, make it stand out in the description (i.e. ``Print the last lines of a given file and keep reading it until `Ctrl + C`:``).

Author: This should be reformatted.
I tested with `grep -r "<<Enter>>"`. This pattern is used nowhere else in the whole project.

Author: See https://github.com/Managor/tldr/blob/style-guide-10/pages/linux/quickemu.md for a place where this needs to be standardised

Reviewer: > This should be reformatted. I tested with `grep -r "<<Enter>>"`. This pattern is used nowhere else in the whole project.

Sure

Author: What do you think, should we use the same syntax in the description as we use in the examples? It would require making sure that they're always in backticks so that they're not invisible to Github markdown rendering, but the consistency is nice for intuitiveness.

- Proper names should be capitalized in the description whenever applicable (e.g. use `A tool for interacting with a Git repository.` instead of ``A tool for interacting with a `git` repository.``).
- Acronym expansions (i.e. protocols, tools, etc) must not be translated unless there is a recognized native equivalent for them.

> - When documenting keycaps or a keyboard shortcut for a utility, use the same [keypress syntax](#keypress-syntax) as is used for example commands. Make sure to enclose it in backtics so that it is not invisible to markdown renderers (i.e. ``Print the last lines of a given file and keep reading it until `<Ctrl c>`:``).

Author: @kbdharun What do you think?

Reviewer: > @kbdharun What do you think?

LGTM, added a minor suggestion.

> - When documenting keycaps or a keyboard shortcut for a utility, use the same [keypress syntax](#keypress-syntax) as is used for example commands. Make sure to enclose it in backtics so that it is not invisible to markdown renderers (i.e. ``Print the last lines of a given file and keep reading it until `<Ctrl c>`:``).

Reviewer: ```suggestion
- When documenting keycaps or a keyboard shortcut for a utility, use the same [keypress syntax](#keypress-syntax) as example commands. Make sure to enclose it in backticks so that it is not invisible to markdown renderers (i.e. ``Print the last lines of a given file and keep reading it until `<Ctrl c>`:``).
```

- Use `{{path/to/source.ext}}` instead of `{{path/to/source.tar[.gz|.bz2|.xz]}}`.

### Keypress syntax

To mark keypresses for TUI or UI programs, use angle brackets `<` and `>`.

- Single characters example: `<a>`.

> - Special keys are to be written with pascal case: `<Ctrl>`, `<Super>`, `<Alt>`, `<Shift>`, `<Cmd>`, `<Option>`, `<Windows>`, `<Enter>`, `<Home>`, `<Space>`, `<Esc>`, `<ArrowUp>`, `<ArrowLeft>`, `<PageUp>`, ...

Author: Standardizing order of simultanous keypress order would be nice.

- Special keys can be translated if they have culturally relevant translations.
- When a program takes in uppercase character literals mark them as `<A>` instead of marking it with shift. Otherwise always mark characters in lowercase.
- Mark simultaneous keypresses inside the same angle brackets separated by a single space:  `<Ctrl c>`, `<Alt F4>`, `<Ctrl Shift k>`, `<Super Shift PrtSc>`.
- Consecutive keypresses need to be contained in their own angle brackets with no space in between: `<:><w><q><Enter>`, `<Ctrl k><Ctrl s>`.

> - Keypresses in the same context or when they form words can be grouped together in the same angle brackets: `<:><help><Enter>`.

Author: Question remain how to make sure that it remains clear that the grouped keypressed are individual keypresses.

Author: Maybe text that is typed out to a prompt, similar to a terminal can be excluded from being marked a keypress? `:` should still be marked as a keypress because it changes context.
```suggestion
- Keypresses in the same context or when they form words can be grouped together in the same angle brackets: `<:>help<Enter>`.
```

- Use `{{path/to/source.ext}}` instead of `{{path/to/source.tar[.gz|.bz2|.xz]}}`.

### Keypress syntax

To mark keypresses for TUI or UI programs, use angle brackets `<` and `>`.

- Single characters example: `<a>`.
- Special keys are to be written with pascal case: `<Ctrl>`, `<Super>`, `<Alt>`, `<Shift>`, `<Cmd>`, `<Option>`, `<Windows>`, `<Enter>`, `<Home>`, `<Space>`, `<Esc>`, `<ArrowUp>`, `<ArrowLeft>`, `<PageUp>`, `<F5>`, `<F12>`, ...
- Special keys can be translated if they have culturally relevant translations.
- When a program takes in uppercase character literals mark them as `<A>` instead of marking it with shift. Otherwise always mark characters in lowercase.
- Mark simultaneous keypresses inside the same angle brackets separated by a single space:  `<Ctrl c>`, `<Alt F4>`, `<Ctrl Shift k>`, `<Super Shift PrtSc>`.
- Consecutive keypresses need to be contained in their own angle brackets with no space in between: `<:><w><q><Enter>`, `<Ctrl k><Ctrl s>`.

> - Keypresses in the same context or when they form words can be grouped together in the same angle brackets: `<:><help><Enter>`.

Author: What to do with grouped keypresses that contain spaces?
![image](https://github.com/user-attachments/assets/e61e8da2-a858-4f3b-ad3a-fbae19440585)

> - Do not place options with multiple choices inside angle brackets: Use `{{<Ctrl Shift PageUp>|<Ctrl Shift PageDown>}}` instead of `<Ctrl Shift {{PageUp|PageDown}}>`.

Author: There needs to be consideration which one of these is clearer 
![image](https://github.com/user-attachments/assets/830a831f-24ec-4954-993b-1233caab9d82)
![image](https://github.com/user-attachments/assets/b9cdd43d-86bc-42f4-9639-5177d5d7ed1e)

Author: Both of these are needed. Removing this as too restrictive.

> - Do not place options with multiple choices inside angle brackets: Use `{{<Ctrl Shift PageUp>|<Ctrl Shift PageDown>}}` instead of `<Ctrl Shift {{PageUp|PageDown}}>`.

Author: Also consider if it's a good idea to put placeholders inside keypress brackets at all 
![image](https://github.com/user-attachments/assets/fdf293d5-8426-4c7b-b0f1-2d25e6042584)

- Use `{{path/to/source.ext}}` instead of `{{path/to/source.tar[.gz|.bz2|.xz]}}`.

### Keypress syntax

To mark keypresses for TUI or UI programs, use angle brackets `<` and `>`.

- Single characters example: `<a>`.

> - Special keys are to be written with pascal case: `<Ctrl>`, `<Super>`, `<Alt>`, `<Shift>`, `<Cmd>`, `<Option>`, `<Windows>`, `<Enter>`, `<Home>`, `<Space>`, `<Esc>`, `<ArrowUp>`, `<ArrowLeft>`, `<PageUp>`, `<F5>`, `<F12>`, ...

Author: Should arrow keys be ArrowUp ArrowDown ArrowLeft ArrowRight or Up Down Left Right?
ArrowDirection is more self evident and just Direction is more condensed.

> - Special keys are to be written with pascal case: `<Ctrl>`, `<Super>`, `<Alt>`, `<Shift>`, `<Cmd>`, `<Option>`, `<Windows>`, `<Enter>`, `<Home>`, `<Space>`, `<Esc>`, `<ArrowUp>`, `<ArrowLeft>`, `<PageUp>`, `<F5>`, `<F12>`, ...

Author: Standardizing simultanous keypress order would be nice.
Shift always after Ctrl, Alt and Super for example.

- Special keys can be translated if they have culturally relevant translations.

> - When a program takes in uppercase character literals mark them as `<A>` instead of marking it with shift. Otherwise always mark characters in lowercase.

Author: Any opinions on this? You always see terminal keypresses marked with uppercase in documentation, but during my brainstorming this made more logical sense.
The lowercase letters do feel less impactful, that's for sure.

Reviewer: IG this is fine, but it might confuse a few new users using tldr but for people used to CLI commands this won't be an issue, also if we added "Shift" and other keys to the example it will increase it's overall size thus making the rendering large too.

- Use `{{path/to/source.ext}}` instead of `{{path/to/source.tar[.gz|.bz2|.xz]}}`.

### Keypress syntax

To mark keypresses for TUI or UI programs, use angle brackets `<` and `>`.

- Single characters example: `<a>`.

> - Special keys are to be written with pascal case: `<Ctrl>`, `<Super>`, `<Alt>`, `<Shift>`, `<Cmd>`, `<Option>`, `<Windows>`, `<Enter>`, `<Home>`, `<Space>`, `<Esc>`, `<ArrowUp>`, `<ArrowLeft>`, `<PageUp>`, ...

Author: ```suggestion
- Special keys are to be written with pascal case: `<Ctrl>`, `<Super>`, `<Alt>`, `<Shift>`, `<Cmd>`, `<Option>`, `<Windows>`, `<Enter>`, `<Home>`, `<Space>`, `<Esc>`, `<ArrowUp>`, `<ArrowLeft>`, `<PageUp>`, ...
```
Should arrow keys be ArrowUp ArrowDown ArrowLeft ArrowRight or Up Down Left Right?

- Use `{{path/to/source.ext}}` instead of `{{path/to/source.tar[.gz|.bz2|.xz]}}`.

### Keypress syntax

> To mark keypresses for TUI or UI programs, use angle brackets `<` and `>`.

Reviewer 2: ```suggestion
To mark keypresses for TUI or GUI programs, use angle brackets `<` and `>`.
```

- Single characters example: `<a>`.

> - Special keys are to be written with [`PascalCase`](https://en.wikipedia.org/wiki/Camel_case): `<Ctrl>`, `<Super>`, `<Alt>`, `<Shift>`, `<Cmd>`, `<Option>`, `<Windows>`, `<Enter>`, `<Home>`, `<Space>`, `<Esc>`, `<ArrowUp>`, `<ArrowLeft>`, `<PageUp>`, `<F5>`, `<F12>`, ...

Reviewer 3: ```suggestion
- Special keys are to be written with [`PascalCase`](https://www.theserverside.com/definition/Pascal-case): `<Ctrl>`, `<Super>`, `<Alt>`, `<Shift>`, `<Cmd>`, `<Option>`, `<Windows>`, `<Enter>`, `<Home>`, `<Space>`, `<Esc>`, `<ArrowUp>`, `<ArrowLeft>`, `<PageUp>`, `<F5>`, `<F12>`, ...
```
Because Pascal case page on Wkipedia redirects to Camel case page

Author: Pascal case is defined on that wikipedia page, but directing to a page that is solely about Pascal case doesn't hurt.

- When writing simultaneous keypresses, keep the following order: `<Ctrl>` -> `<Super>` -> `<Alt>` -> `<AltGr>` -> `<Shift>` -> everything else.

> - Special keys can be translated if they have culturally relevant translations.

Author: What do the parentheses clarify? What other options are there even?

> - Special keys can be translated if they have culturally relevant translations.

Author: Could we rather change the linter to allow translations inside `<>`? I would rather not place unnecessary placeholders inside the commands.

> - Special keys can be translated if they have culturally relevant translations.

Reviewer: > Could we rather change the linter to allow translations inside `<>`? I would rather not place unnecessary placeholders inside the commands.

That would work too, feel free to open an issue at https://github.com/tldr-pages/tldr-lint/issues
